create map structuring

FJr-Unifil opened the pull request:
- **Add: mapstruct dependencies and lombok compatible version**
- **Implement clinic MapStruct**
- **Implement account planning MapStruct**
- **Implement involved MapStruct**
- **Refactor Clinic isActive property: Boolean -> boolean**
- **Refactor clinic MapStruct**

FJr-Unifil committed:
Add: mapstruct dependencies and lombok compatible version

FJr-Unifil committed:
Implement clinic MapStruct

FJr-Unifil committed:
Implement account planning MapStruct

FJr-Unifil committed:
Implement involved MapStruct

FJr-Unifil committed:
Refactor Clinic isActive property: Boolean -> boolean

This changes fixes inconsistencies between the Lombok getter and setter.

FJr-Unifil committed:
Refactor clinic MapStruct

pietroow commented on src/main/java/example/medCashFlow/model/Clinic.java:
```diff
@@ -31,14 +31,9 @@ public class Clinic {
     private String phone;
 
     @Column(name = "is_active", nullable = false)
```
como boolean nativo nao permite nulo e a coluna esta como nullable false
tambem seria legal a gente colocar um defaultValue para esse campo

FJr-Unifil committed:
Implement bills MapStruct

FJr-Unifil committed:
Implement employees MapStruct

FJr-Unifil committed:
Refactor: eliminating unused employee constructor

FJr-Unifil commented on src/main/java/example/medCashFlow/model/Clinic.java:
```diff
@@ -31,14 +31,9 @@ public class Clinic {
 
     @Column(name = "is_active", nullable = false)
-    private Boolean isActive = true;
+    private boolean isActive = true;
```
No caso isso não conta como default value? Seria somente no arquivo SQL? Qual a boa prática para casos assim?

pietroow replied:
ate conta, mas a ideia seria colocar na migration do banco de dados que essa coluna tem um valor default...
caso alguem tente fazer um insert direto no banco de dados e nao usar a api por exemplo.... a gente ainda continua com o dado valido na base de dados

FJr-Unifil committed:
Fix: mapPassword to manager

pietroow commented on src/main/java/example/medCashFlow/controller/AccountPlanningController.java:
```diff
@@ -53,11 +51,7 @@ public ResponseEntity<AccountPlanningResponseDTO> createAccountPlanning(
-        newAccountPlanning.setDescription(data.description());
-        newAccountPlanning.setEmoji(data.emoji());
-        newAccountPlanning.setColor(data.color());
+        AccountPlanning newAccountPlanning = accountPlanningService.toAccountPlanning(data);
```
accountPlanningService.toAccountPlanning(data, employee.getClinic() );

pietroow commented on src/main/java/example/medCashFlow/services/AccountPlanningService.java:
```diff
@@ -15,26 +17,37 @@
+        return mapper.toResponseDTO(accountPlanning);
+    }
+
+    public AccountPlanning toAccountPlanning(AccountPlanningRegisterDTO data) {
```
```suggestion
    public AccountPlanning toAccountPlanning(AccountPlanningRegisterDTO data, Clinic clinic) {
```

pietroow commented on src/main/java/example/medCashFlow/services/AccountPlanningService.java:
```diff
@@ -15,26 +17,37 @@
+    }
+
+    public AccountPlanning toAccountPlanning(AccountPlanningRegisterDTO data) {
+        return mapper.toAccountPlanning(data);
```
```suggestion
        return mapper.toAccountPlanning(data, clinic);
```

pietroow commented on src/main/java/example/medCashFlow/mappers/AccountPlanningMapper.java:
```diff
@@ -0,0 +1,18 @@
+
+    @Mapping(target = "id", ignore = true)
+    @Mapping(target = "clinic", ignore = true)
+    AccountPlanning toAccountPlanning(AccountPlanningRegisterDTO accountPlanningRegisterDTO);
```
```suggestion
    AccountPlanning toAccountPlanning(AccountPlanningRegisterDTO accountPlanningRegisterDTO, Clinic clinic);
```

FJr-Unifil committed:
Refactor: update EmployeeMapper

FJr-Unifil committed:
Refactor: update InvolvedMapper

FJr-Unifil committed:
Refactor: update AccountPlanningMapper

FJr-Unifil committed:
Refactor: update BillMapper

FJr-Unifil committed:
Refactor: removed unused imports

FJr-Unifil committed:
Refactor employee structure

Change the DTO structure, by making it more clean and fixing the mapper being
overload by what should be the service responsibility

FJr-Unifil committed:
Refactor: employeeService removing duplicate methods

FJr-Unifil committed:
Refactor bill structure

Change the DTO structure, by making it more clean and fixing the mapper being
overload by what should be the service responsibility

FJr-Unifil committed:
Refactor accountPlanning structure

Change the controller to just handle the request and delegating the
responsibility of manipulating business logic to service

FJr-Unifil committed:
Refactor involved structure

Change the controller to just handle the request and delegating the
responsibility of manipulating business logic to service

FJr-Unifil committed:
Refactor clinic structure

Change the controller to just handle the request and delegating the
responsibility of manipulating business logic to service

pietroow commented on src/main/java/example/medCashFlow/mappers/EmployeeMapper.java:
```diff
@@ -0,0 +1,32 @@
+    EmployeeResponseDTO toResponseDTO(Employee employee);
+
+    @Named("mapPassword")
+    default String mapPassword(String password) {
```
colocar a logica na service

pietroow commented on src/main/java/example/medCashFlow/services/BillService.java:
```diff
@@ -31,31 +41,23 @@ public List<BillResponseDTO> getAllBillsByClinicId(UUID clinicId) {
-        existingBill.setDueDate(data.dueDate());
-        existingBill.setInstallmentsAmount(data.installments());
+        Bill updatedBill = toBill(data, employee);
+        updatedBill.setId(id);
```
sem necessidade

pietroow commented on src/main/java/example/medCashFlow/services/EmployeeService.java:
```diff
@@ -15,34 +19,32 @@
 
     public boolean isEmployeeValid(String cpf, String email) {
-        return (isEmployeeValidByCpf(cpf) && isEmployeeValidByEmail(email));
+        return (!isEmployeeValidByCpf(cpf) || !isEmployeeValidByEmail(email));
```
```suggestion
        return !(isEmployeeValidByCpf(cpf) || isEmployeeValidByEmail(email));
```

VALIDAR COM O GEPETO

FJr-Unifil replied:
Não é a mesma coisa, para ser a mesma coisa deveria ser: `return (!A && !B)`, no caso acima está como `return (!A || !B). Porém eu melhorei a forma para ler a condição, deixando de mais fácil entendimento.

pietroow commented on src/main/java/example/medCashFlow/services/EmployeeService.java:
```diff
@@ -61,55 +63,36 @@ public boolean isEmployeeValidByEmail(String email) {
 
-        if (!employee.getCpf().equals(existingEmployee.getCpf())
-                && repository.existsByCpf(employee.getCpf())) {
+        if (!data.cpf().equals(existingEmployee.getCpf())
```
findByCpfAndIdNot(cpf, idTobeIgnored)

pietroow replied:
analisar a possibilidade de colocar uma query pra validar os dois casos

FJr-Unifil committed:
Refactor authController with service responsibility

FJr-Unifil committed:
Refactor encrypting password to service

FJr-Unifil committed:
Implement employee update mapping

FJr-Unifil committed:
Implement bill update mapping

FJr-Unifil committed:
Implement involved update mapping

FJr-Unifil committed:
Implement accountPlanning update mapping

FJr-Unifil committed:
Refactor boolean condition for better readability

FJr-Unifil committed:
Refactor boolean condiiton to be a repository query

FJr-Unifil committed:
Refactor: create dedicated BillDependencies DTO

FJr-Unifil committed:
Refactor: using @ResponseStatus when thre is no response body

FJr-Unifil committed:
Merge with main
